pennylane/resource/specs.py

```diff
@@ -272,22 +511,171 @@ def circuit():
+
+        *Runtime resource tracking* (specified by ``level="device"``) works by mock-executing the desired
+        workflow and tracking the number of times a given gate has been applied. This mock-execution happens
+        after all compilation steps, and should be highly accurate to the final gatecounts of running on
+        a real device.
+
+        .. code-block:: python
+
+            qml.capture.enable()  # Enable program capture to allow these transforms to be applied only as MLIR passes
+
+            dev = qml.device("lightning.qubit", wires=3)
+
+            @qml.qjit
+            @qml.transforms.merge_rotations
+            @qml.transforms.cancel_inverses
+            @qml.qnode(dev)
+            def circuit(x):
+                qml.RX(x, wires=0)
+                qml.RX(x, wires=0)
+                qml.X(0)
+                qml.X(0)
+                qml.CNOT([0, 1])
+                return qml.probs()
+
+        ::
+
+            >>> print(qml.specs(circuit, level="device")(1.23))
+            Device: lightning.qubit
+            Device wires: 3
+            Shots: Shots(total=None)
+            Level: device
+            <BLANKLINE>
+            Resource specifications:
+              Total qubit allocations: 3
+              Total gates: 2
+              Circuit depth: 2
+            <BLANKLINE>
+              Gate types:
+                CNOT: 1
+                RX: 1
+            <BLANKLINE>
+              Measurements:
+                No measurements.
+
+        .. warning::
+            Measurement data is not currently supported with runtime resource tracking, so measurement
+            data may show as missing.
+
+        *Pass-by-pass specs* can be obtained by analyzing the intermediate representations of compiled circuits.
+        This can be helpful for determining how circuit resources change after a given transform
+        or compilation pass.
+        This version of specs can be applied by passing one of the following values for the `level` argument:
+
+        * An ``int``: the desired pass level of a user-applied pass, see the note below
+        * A marker name (str): The name of an applied :func:`qml.marker <pennylane.marker>` pass
+        * An iterable: A list, tuple, etc. containing ints and/or marker names. Should be sorted in
+          ascending pass order
+        * The string "all": To output information about all user-applied transforms and compilation passes
+        * The string "all-mlir": To output information about all compilation passes at the MLIR level only
+
+        .. note::
+            The level arguments only take into account user-applied transforms and compilation passes.
+            Level 0 always corresponds to the original circuit before any user transforms have been applied,
+            and incremental levels correspond to the order in which user transforms were applied.
+            The following level is the MLIR lowering pass, followed by any MLIR-level compilation passes.
```

Comment:
```suggestion
            and incremental levels correspond to the aggregate of user transforms in the order in which they were applied.
            
            When a compilation pass is encountered that is applied in MLIR by Catalyst, ``specs`` will show a "lowering" pass that takes place for translating a program into MLIR. This only occurs with PennyLane's program capture disabled (it is disabled by default with Catalyst). 
```

How's this?

Reply:
The lowering pass is always one of the possible levels for this kind of specs. It doesn't depend on program capture (although for some passes, enabling/disabling program capture can change where in the pipeline this lowering pass happens). If there are no compilation passes applied then the lowering pass will show as the last pass

Reply:
```suggestion
            and incremental levels correspond to the aggregate of user transforms in the order in which they were applied.
            
            In addition, ``"all"`` may or may not show an MLIR "lowering" pass that indicates that the program had to be lowered into MLIR for further compilation with Catalyst.
```

How about we just keep this simple then?